src/GithubApi.hs

```diff
@@ -63,3 +77,30 @@ runGithub auth projectInfo operation =
         Left err -> logWarnN $ Text.append "Failed to comment: " $ Text.pack $ show err
         Right _ -> logInfoN $ Text.concat ["Posted comment on ", Text.pack $ show pr, ": ", body]
       pure cont
+
+    HasPushAccess (Username username) cont -> do
+      result <- liftIO $ Github3.collaboratorPermissionOn
+        (Just auth)
+        (Github3.N $ Project.owner projectInfo)
+        (Github3.N $ Project.repository projectInfo)
+        (Github3.N username)
+
+      case result of
+        Left err -> do
+          logErrorN
+            $ Text.append "Failed to retrive collaborator status: "
+            $ Text.pack $ show err
+          -- To err on the safe side, if the API call fails, we pretend nobody
+          -- has push access.
+          pure $ cont False
```

Comment:
If you make this a three-valued enum instead of a Boolean then you can make an interpreter that retries this operation.

Reply:
Probably we would want something that can retry all GitHub calls, not just this one? And then it should apply exponential backoff and possibly also handle rate limits. But I think for now we can get away with not retrying anything, the call volume will be pretty low, and if it fails at some point, we can just leave a new comment.